app.py

```diff
@@ -755,7 +770,15 @@ def newlabel():
 	"""Re-draw tree with newly picked label."""
 	data = request.get_json()
 	treestr = data.get('tree')
-	treeobj, cgel_tree_terminals = graphical_operation_preamble(treestr)
+	try:
+		treeobj, cgel_tree_terminals = graphical_operation_preamble(treestr)
+	except Exception as err:
+		if len(str(err)) > 0:
+			error_msg = str(err)
+		else:
+			error_msg = "Your CGEL tree is structurally deficient."
+		msg = "ERROR: " + error_msg + "\n\nPlease correct tree errors before proceeding."
```

Comment:
repeating the same logic? can it be factored out?

Reply:
Refactored in https://github.com/nschneid/activedop/pull/105/commits/5b0f2980caa8fbb0ad2e4f997582707253b47c3f. (Raise error in the `ActivedopTree` class method called by the repeating lines).